cp_core/src/task/transform/common.rs

```diff
@@ -9,33 +11,31 @@ use crate::{
     util::error::{CpError, CpResult},
 };
 
-use super::config::RootTransformConfig;
-
 /// Base transform trait. Takes
 pub trait Transform {
     fn run(&self, main: LazyFrame, ctx: Arc<DefaultPipelineContext>) -> CpResult<LazyFrame>;
 }
 
-pub trait SubTransformConfig {
+pub trait TransformConfig {
     fn validate(&self) -> Vec<CpError>;
-    fn transform(self) -> Box<dyn Transform>;
+    fn transform(&self) -> Box<dyn Transform>;
```

Comment:
lmk what is the issue with having it as `transform(self)` and i'll see if we can revert it back, if not idt its a big issue to keep it as ref either.

Reply:
![image](https://github.com/user-attachments/assets/1961f456-4a72-4092-bba1-e2092157c56a)
tbh I don't really understand rust enough to make sense of why we cannot move this just because it's dynamic dispatch and unknown at runtime.

Reply:
i see... ok then i think keep to the current way (`transform(&self)`)